### PR TITLE
migrate errorHandler tests from Jest to Vitest 

### LINKED ISSUE #2758: Refactor: `src/utils/errorHandler.test.tsx` from Jest to Vitest
**Is your feature request related to a problem? Please describe.**
This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor this file's test cases to align with Vitest's syntax and features.

**Acceptance Criteria**
1. Replace Jest-specific functions and mocks with Vitest equivalents
1. Rename the test file from having a `.test.*` suffix to  a `.spec.*` suffix
1. Ensure all tests in the file pass after migration using `npm run test:vitest`
1. Maintain the test coverage for the file as 100% after migration
1. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation:
https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Dhiren-Mhatre**
**What kind of change does this PR introduce?**
Refactor: Migration from Jest to Vitest for error handler test files

**Issue Number:**
Fixes #2758

**Did you add tests for your changes?**
Yes. The migrated tests maintain 100% coverage and all tests are passing.

**Snapshots/Videos:**
 

https://github.com/user-attachments/assets/b7f7ab57-2769-49e1-b7d8-617a2049dc87

**If relevant, did you update the documentation?**
N/A - This is a test file migration

**Summary**
This PR migrates error handler test files from Jest to Vitest as part of our ongoing migration to Vitest.

Key changes:
- Migrated error handler test files to Vitest syntax
- Renamed test files to follow .spec.ts convention
- Replaced Jest-specific functions with Vitest equivalents
- Added proper TypeScript interfaces where needed
- Maintained 100% test coverage

**Does this PR introduce a breaking change?**
No
 
**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **Tests**
	- Enhanced unit tests for the Organization Dashboard component, improving comment clarity and simplifying assertions.
	- Introduced comprehensive unit tests for the errorHandler function, covering various error scenarios and edge cases.

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
You haven't deleted the orginal test file as requested

File: src/screens/OrganizationDashboard/OrganizationDashboard.spec.tsx [hunk 1]
```diff
@@ -160,12 +160,8 @@ describe('Testing Organization Dashboard Screen', () => {
...
     expect(screen.getByText(t.blockedUsers)).toBeInTheDocument();
 
-    // Upcoming events - Using more flexible matcher
-    await waitFor(() => {
-      expect(
-        screen.getByText(/Event 1/i, { exact: false }),
-      ).toBeInTheDocument();
-    });
+    // Upcoming events - Use a more flexible matcher
+    expect(screen.getByText(/Event 1/i, { exact: false })).toBeInTheDocument();
```
Comment (palisadoes) [inline src/screens/OrganizationDashboard/OrganizationDashboard.spec.tsx:164]: Why was this edited?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments on editing
- src/screens/OrganizationDashboard/OrganizationDashboard.spec.tsx

File: src/screens/OrganizationDashboard/OrganizationDashboard.spec.tsx (hunk 1 above)
Comment (Dhiren-Mhatre) [reply src/screens/OrganizationDashboard/OrganizationDashboard.spec.tsx:164]: I happened to see a failing test in src/screens/OrganizationDashboard/OrganizationDashboard.spec.tsx. I also asked this question on slack general channel wher peter harrison said "Please fix the test if it is causing other tests to fail".
Screenshot from 2024-12-26 18-50-58

**Review by palisadoes — APPROVED** [review]

**Pull request merged**